### PR TITLE
Customize journey page

### PR DESCRIPTION
**Pull request by ak47na**
For each journey, displays two buttons:
1. `mark journey as complete` -> which deletes the journey from the page
2. `view journey on map`: creates a Google map, adds the waypoints of the journey as markers and a polyline between them

A future PR will delete the journey also from the datastore when it is marked completed and will add details for each marker on the map.

### CODE REVIEW

**Review by anamariamilcu — APPROVED** [review]

File: src/main/webapp/journeys.html [hunk 1]
```diff
@@ -12,16 +12,19 @@
     <script src="common.js"></script>
     <script src="app.js"></script>
     <link href="style.css" rel="stylesheet" type="text/css" media="screen" />
+    <script src="mapsScript.js"></script>
   </head>
   
-  <body onload="displayJourneys()">
-
+  <body onload="loadMapsScript();displayJourneys()">
```
Comment (ghost) [inline src/main/webapp/journeys.html]: You might want to consider using `DOMContentLoaded` - `onload` waits for *everything* including images and stylesheets etc, but `DOMContentLoaded` waits only for the DOM and scripts.

```html
<script>
  document.addEventListener('DOMContentLoaded', () => {
    loadMapsScript();
    displayJourneys();
  });
</script>


```

File: src/main/webapp/journeys.html [hunk 2]
```diff
@@ -12,16 +12,19 @@
...
     <div class="menu-container"></div>
 
     <div class="content-container">
-      <h1>Your delivery journeys: </h1>
-      <ul id="journeys-container"></ul>
+      <div
```
Comment (ghost) [inline src/main/webapp/journeys.html]: Missing `>`?
```suggestion
      <div> 
```

File: src/main/webapp/mapsScript.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+ */
+
+/**
+ * Fetch apiKey and dynamically load the Maps JavaScript API.
+ */
+function loadMapsScript() {
+  fetch('apiKey.json').then(response => response.json()).then(jsonResponse => {
+    GoogleMapsApiKey = jsonResponse.apiKey;
+    // Create the Google Maps tag with the fetched api key.
+    let script = document.createElement('script');
```
Comment (ghost) [inline src/main/webapp/mapsScript.js]: Generally speaking we like to use `const` if we never need to reassign `script`.

it is a good habit to get into always using `const` by default, and only using `let` as an exception.

```suggestion
    const script = document.createElement('script');
```

File: src/main/webapp/mapsScript.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+/**
+ * Creates and returns a Map centered at (centerLat, centerLng). 
+ */
+function initMap(centerLat, centerLng) {
+  const map = new google.maps.Map(
+      document.getElementById('map'),
+      {center: {lat: centerLat, lng: centerLng}, zoom: 12});
+  return map;
```
Comment (ghost) [inline src/main/webapp/mapsScript.js]: Might be worth allowing for an optional zoom level? You can provide a default value here if the caller does not provide one.

```suggestion
function initMap(centerLat, centerLng, zoomLevel = 12) {
  const map = new google.maps.Map(
      document.getElementById('map'),
      {center: {lat: centerLat, lng: centerLng}, zoom: zoomLevel});
  return map;
```

File: src/main/webapp/mapsScript.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  numberOfWaypoints = waypoints.length;
+  for (const waypoint in waypoints) {
+    centerLat += waypoints[waypoint].point.latitude;
+    centerLng += waypoints[waypoint].point.longitude;
+  }
+
+  let waypointsCoordinates = [];
+  map = initMap(centerLat / numberOfWaypoints, centerLng / numberOfWaypoints);
+  for (const waypoint in waypoints) {
+    waypointCoordinates = {lat: waypoints[waypoint].point.latitude, lng: waypoints[waypoint].point.longitude};
```
Comment (ghost) [inline src/main/webapp/mapsScript.js]: Optional formatting suggestion for code clarity:

```suggestion
    waypointCoordinates = {
      lat: waypoints[waypoint].point.latitude,
      lng: waypoints[waypoint].point.longitude,
    };
```

File: src/main/webapp/mapsScript.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    // Add the current waypoint to the map.
+    const waypointMarker = new google.maps.Marker({
+    position: waypointCoordinates,
+    map: map,
+    title: `waypoint${waypoint+1}`
+  });
```
Comment (ghost) [inline src/main/webapp/mapsScript.js]: Optional indentation formatting suggestion for code clarity:
```suggestion
    const waypointMarker = new google.maps.Marker({
      position: waypointCoordinates,
      map: map,
      title: `waypoint${waypoint+1}`
    });
```

File: src/main/webapp/mapsScript.js [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+function addPolyLineBetweenWaypoints(map, waypointsCoordinates) {
+  const journeyPath = new google.maps.Polyline({
+    path: waypointsCoordinates,
+    geodesic: true,
+    strokeColor: "#FF0000",
+    strokeOpacity: 1.0,
+    strokeWeight: 2
+  });
+  journeyPath.setMap(map);
+}
```
Comment (ghost) [inline src/main/webapp/mapsScript.js]: Optional: you could also do default parameters (like in the above comment) here for the stroke specifics (color, opacity, weight)

**Review by ghost — APPROVED** [review]

**Review by anamariamilcu — APPROVED** [review]

File: src/main/webapp/journeys.html (hunk 1 above)
Comment (ak47na) [reply src/main/webapp/journeys.html]: Right, done.

File: src/main/webapp/mapsScript.js (hunk 3 above)
Comment (ak47na) [reply src/main/webapp/mapsScript.js]: Thanks, I changed it to const.

File: src/main/webapp/mapsScript.js (hunk 4 above)
Comment (ak47na) [reply src/main/webapp/mapsScript.js]: Yes, I added it.

File: src/main/webapp/mapsScript.js (hunk 6 above)
Comment (ak47na) [reply src/main/webapp/mapsScript.js]: Done.

File: src/main/webapp/mapsScript.js (hunk 7 above)
Comment (ak47na) [reply src/main/webapp/mapsScript.js]: Yes, I added color, opacity, weight as optional parameters.

**Pull request merged**